pkg/controller/shared_state.go

```diff
@@ -1252,13 +1254,12 @@ func (c *SharedState) manageReplicas(ctx context.Context, volumeName string) err
 		return nil
 	}
 
-	azVolumeAttachments, err := azureutils.GetAzVolumeAttachmentsForVolume(ctx, c.cachedClient, volumeName, azureutils.ReplicaOnly)
-	if err != nil {
-		w.Logger().Errorf(err, "failed to list replica AzVolumeAttachments")
+	currentReplicaCount := 0
+	if currentReplicaCount, err = c.countValidReplicasForVolume(ctx, volumeName); err != nil {
```

Comment:
np: move `currentReplicaCount, err = c.countValidReplicasForVolume(ctx, volumeName)` outside of the if scope to eliminate the unnecessary initialization of `currentReplicaCount:=0`

Reply:
But we use the same `err` variable in this method for defer function `w.Finish(err)`. I still need to create and initialize `currentReplicaCount` variable first even if `countValidReplicasForVolume` is moved outside of `if` scope. Is it right?

Reply:
You're right. Create a new `err` variable shouldn't affect the workflow to call `Finish(err)`.